### DISCUSSION TITLE
Paginate count(*) query issue.

### DISCUSSION THREAD

**Discussion by nikunj-7span** [laravel/framework#45564] [Q&A]
- Laravel Version: 8.0
- PHP Version: 7.3.0
- Database Driver & Version:  MySQL 5.7.34

### Describe the Bug
When I use paginate for pagination the data it call the count( * ) query every time even I pass the column name like count(['id']), and the count( * ) query scan all row in the table. 

- Table name is `mobiles` and it has 45 column 

- Route  
```php 
Route::get("mobiles", "Api\V1\MobileController@show");

```

- Controller 
```php
namespace App\Http\Controllers\Api\V1;

class MobileController extends Controller
{
    public function show()
    {
        return Mobile::paginate(10 , ['id']);
    }
 }
```
- Call `mobiles` route 
 - Telescope showing me that two queries
###  Actual result

  <img width="982" alt="image" src="https://user-images.githubusercontent.com/92076426/211253715-9f552bcb-186c-44cb-b8a2-2686fa2d07c8.png">

### Expected  result 

  <img width="953" alt="image" src="https://user-images.githubusercontent.com/92076426/211254016-2b35ad0e-ee3a-4eeb-9896-fe736f351073.png">
  
### Steps To Reproduce:

<img width="1106" alt="image" src="https://user-images.githubusercontent.com/92076426/211250400-562ef8d6-1aec-4de9-94e1-dc3adb66e30c.png">

- The following image shows that I had done changes as per our functionality, It will take the first column name from the passed in the paginate method array of $columns params and that query does not scan all columns of the mobiles tables.

### Final Results: 
<img width="953" alt="image" src="https://user-images.githubusercontent.com/92076426/211254016-2b35ad0e-ee3a-4eeb-9896-fe736f351073.png">

**Comment by parallels999** [comment]
Sounds like a great addition, did you make a pull request?
